mpmp/utilities/file_utilities.py

```diff
@@ -59,23 +114,32 @@ def save_results(output_dir,
         float_format="%.5g"
     )
 
-    output_file = Path(
-        output_dir, "{}_{}_{}_s{}_auc_threshold_metrics.tsv.gz".format(
-            identifier, training_data, signal, seed)).resolve()
+    output_file = construct_filename(output_dir,
+                                     identifier,
+                                     shuffle_labels,
+                                     model_options,
+                                     'auc_threshold_metrics',
+                                     '.tsv.gz')
     auc_df.to_csv(
         output_file, sep="\t", index=False, compression="gzip", float_format="%.5g"
```

Comment:
I think pandas automatically detects the compression algorithm from the file extension provided (`.tar.gz`). If this is true, it might be better to avoid specifying `compression="gzip"` here, and just handle that using the file extension.

Reply:
Cool, seems like this works! I didn't know pandas could infer the compression algorithm automatically.